pkg/splunk/enterprise/afwscheduler.go

```diff
@@ -273,12 +281,160 @@ func getPhaseInfoByPhaseType(worker *PipelineWorker, phaseType enterpriseApi.App
 	return &worker.appDeployInfo.PhaseInfo
 }
 
+// updatePplnWorkerPhaseInfo updates the in-memory PhaseInfo
+func updatePplnWorkerPhaseInfo(appDeployInfo *enterpriseApi.AppDeploymentInfo, phaseType enterpriseApi.AppPhaseType, retryCount int32, statusType enterpriseApi.AppPhaseStatusType) {
+	scopedLog := log.WithName("updatePplnWorkerPhaseInfo").WithValues("appName", appDeployInfo.AppName)
+
+	scopedLog.Info("changing the status", "old status", appPhaseStatusAsStr(appDeployInfo.PhaseInfo.Status), "new status", appPhaseStatusAsStr(statusType))
+	appDeployInfo.PhaseInfo.Phase = phaseType
+	appDeployInfo.PhaseInfo.RetryCount = retryCount
+	appDeployInfo.PhaseInfo.Status = statusType
+}
```

Comment:
Ya, with respect to the status it is fine. But,  the idea is, it should not touch the `Phase`. I know, here you are writing the same phase again, but we don't want to give that provision, just to avoid any misuses in future.